### ISSUE TITLE
AspNetIdentity not usable with IDSvr4 2.0 preview due to breaking namespace change in Microsoft.AspNetCore.Builder

### ISSUE THREAD

**Issue by aluethi** [IdentityServer/IdentityServer4.AspNetIdentity#33]
Calling AddAspNetIdentity<>() in ConfigureServices method of a Startup.cs results in the following exception being thrown (btw: I'm trying to use the 2.0.0-preview3 of IdentityServer4 with the 1.0.1 version of IdentityServer4.AspNetIdentity):

`Could not load type 'Microsoft.AspNetCore.Builder.IdentityOptions' from assembly 'Microsoft.AspNetCore.Identity, Version=2.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.`

I believe this is related to the following issue in Microsoft.AspNetCore.Identity referenced here: Move IdentityOptions from Microsoft.AspNetCore.Builder to identity namespace #1261 (https://github.com/aspnet/Identity/issues/1261)

Do you think a change in the used namespace will remedy this or are there other dependencies I'm not aware of?

**Comment by leastprivilege** [comment]
Yep - it's not done yet.

https://github.com/IdentityServer/IdentityServer4/issues/1055#issuecomment-324819806

**Comment by rizamarhaban** [comment]
I found the breaking code on the IdentityServer4.AspNetIdentity on this file: IdentityServerBuilderExtensions.cs, I try looking for the changes or extensions on ASP.NET Core 2.0 but cannot find one as well.

![image](https://user-images.githubusercontent.com/1063940/29863526-8ab5a7f4-8da2-11e7-9733-4160d6af427f.png)

**Comment by rizamarhaban** [comment]
Hey it works,

Here is what I did on the ```IdentityServer4.AspNetIdentity``` fork (I know probably this is not the right way, but for my demo its just works perfectly).

First on ```IdentityServerBuilderExtensions.cs``` file, I commented the lines:
![image](https://user-images.githubusercontent.com/1063940/29865166-6ff78c70-8da7-11e7-93b1-2cde087131d2.png)

Second on the ```ProfileServices.cs``` I replace the ```AddFilterClaims``` with ```AddRequestedClaims``` method. There is a ```FilterClaims``` extension by the way. But don't know how to implement it.
![image](https://user-images.githubusercontent.com/1063940/29865266-acfd212a-8da7-11e7-9d03-ee7630e0d674.png)

Compile it and I can use with ASP.NET Core 2.0. I haven't test using Claims, probably will get an error.
